### PR TITLE
fix(curriculum): clean-up Project Euler 201-220

### PR DESCRIPTION
**Pull request by gikf**
- Cleans description.
- Changes generic `eulerXXX` function name.
- Adds images.
- Uses MathJax, a lot.
- Related to #40896.

Needed before merging:
- [x] Merged PR adding needed images to CDN https://github.com/freeCodeCamp/cdn/pull/83.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-217-balanced-numbers.md [hunk 1]
```diff
@@ -8,35 +8,35 @@ dashedName: problem-217-balanced-numbers
 
 # --description--
 
-A positive integer with k (decimal) digits is called balanced if its first ⌈k/2⌉ digits sum to the same value as its last ⌈k/2⌉ digits, where ⌈x⌉, pronounced ceiling of x, is the smallest integer ≥ x, thus ⌈π⌉ = 4 and ⌈5⌉ = 5.
+A positive integer with $k$ (decimal) digits is called balanced if its first $⌈\frac{k}{2}⌉$ digits sum to the same value as its last $⌈\frac{k}{2}⌉$ digits, where $⌈x⌉$, pronounced ceiling of $x$, is the smallest integer $≥ x$, thus $⌈π⌉ = 4$ and $⌈5⌉ = 5$.
 
 So, for example, all palindromes are balanced, as is 13722.
 
-Let T(n) be the sum of all balanced numbers less than 10n.
+Let $T(n)$ be the sum of all balanced numbers less than $10n$.
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-217-balanced-numbers.md:15]:
```suggestion
Let $T(n)$ be the sum of all balanced numbers less than $10^n$.
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-219-skew-cost-coding.md [hunk 2]
```diff
@@ -8,39 +8,39 @@ dashedName: problem-219-skew-cost-coding
...
+A prefix-free code of size $n$ is a collection of $n$ distinct bit strings such that no string is a prefix of any other. For example, this is a prefix-free code of size 6:
 
-0000, 0001, 001, 01, 10, 11
+$$0000, 0001, 001, 01, 10, 11$$
 
-Now suppose that it costs one penny to transmit a '0' bit, but four pence to transmit a '1'. Then the total cost of the prefix-free code shown above is 35 pence, which happens to be the cheapest possible for the skewed pricing scheme in question. In short, we write Cost(6) = 35.
+Now suppose that it costs one penny to transmit a '0' bit, but four pence to transmit a '1'. Then the total cost of the prefix-free code shown above is 35 pence, which happens to be the cheapest possible for the skewed pricing scheme in question. In short, we write $Cost(6) = 35$.
 
-What is Cost(109) ?
+What is $Cost(109)$?
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-219-skew-cost-coding.md:23]:
```suggestion
What is $Cost(10^9)$?
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-220-heighway-dragon.md [hunk 3]
```diff
@@ -8,37 +8,46 @@ dashedName: problem-220-heighway-dragon
...
+- "a" → "aRbFR"
+- "b" → "LFaLb"
 
-Thus, D0 = "Fa", D1 = "FaRbFR", D2 = "FaRbFRRLFaLbFR", and so on.
+Thus, $D_0$ = "Fa", $D_1$ = "FaRbFR", $D_2$ = "FaRbFRRLFaLbFR", and so on.
 
 These strings can be interpreted as instructions to a computer graphics program, with "F" meaning "draw forward one unit", "L" meaning "turn left 90 degrees", "R" meaning "turn right 90 degrees", and "a" and "b" being ignored. The initial position of the computer cursor is (0,0), pointing up towards (0,1).
 
-Then Dn is an exotic drawing known as the Heighway Dragon of order n. For example, D10 is shown below; counting each "F" as one step, the highlighted spot at (18,16) is the position reached after 500 steps.
+Then $D_n$ is an exotic drawing known as the Heighway Dragon of order n. For example, $D_{10}$ is shown below; counting each "F" as one step, the highlighted spot at (18,16) is the position reached after 500 steps.
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-220-heighway-dragon.md:20]:
```suggestion
Then $D_n$ is an exotic drawing known as the Heighway Dragon of order $n$. For example, $D_{10}$ is shown below; counting each "F" as one step, the highlighted spot at (18,16) is the position reached after 500 steps.
```

**Review by moT01 — COMMENTED** [review]
👍

**Review by moT01 — APPROVED** [review]
LGTM 🎉 assuming tests pass

**Review by majestic-owl448 — APPROVED** [review]
Everything seems fine

LGTM

**Pull request merged**